components/items/ItemsGrid/useNftActions.ts

```diff
@@ -0,0 +1,187 @@
+import { isOwner as checkOwner } from '@/utils/account'
+import nftById from '@/queries/subsquid/general/nftById.graphql'
+import listCount from '@/queries/subsquid/general/countOfTokenNftsToList.graphql'
+import nftListWithSearch from '@/queries/subsquid/ahk/nftListWithSearch.graphql'
+import { TokenId } from '~/components/rmrk/service/scheme'
+
+export type NFTWitToken = NFTWithMetadata & TokenId
+export const isStack = (entity: TokenEntity) => entity.supply > 1
+
+export const userIsOwner = (entity: TokenEntity, accountId: string) =>
+  checkOwner(entity.cheapest?.currentOwner, accountId)
+
+export const isAvailableToBuy = (entity: TokenEntity) =>
+  Number(entity.cheapest.price) > 0
+
+const tokensNftsCache = new Map<string, NFTWithMetadata[]>()
+
+export const fetchNft = async (nftId: string): Promise<NFTWithMetadata> => {
```

Comment:
useAsyncData?

Reply:
ill check

Reply:
I think you are referring to `useAsyncData` caching
1. I didn't know - thanks
2. not good enough here: in `itemsGrid` there is a single call to get all of the nfts of the tokens that _might_ get listed later
but each token's nfts are cached individually

the result is when user click `list` on a token that has already been fetched it isn't fetched again

to achieve that using `useAsyncData` ill need to split the initial query up, each token to its own query